packages/core/src/StateNode.ts

```diff
@@ -354,6 +354,17 @@ export class StateNode<
     state: State<TContext, TEvent>,
     _event: SCXML.Event<TEvent>
   ): StateTransition<TContext, TEvent> | undefined {
+    if (this.type === 'history' && state.historyMap[this.id]) {
```

Comment:
I'm not entirely sure yet, but this might introduce a regression - entering remembered history states is still entering (when we look at SCXML they are just added to `entrySet` in `computeEntrySet` & later on are treated as any other states in `enterStates` which executes executable content). And judging by this returned value & the algorithm of `getActions` I believe those entry actions wouldnt get executed.

As there were no tests for this behavior I think it's OK to let it slip for now, we should just be aware of that and fix it soon - I'm touching the whole transition resolving algorithm in https://github.com/davidkpiano/xstate/pull/825 and I think I might be able to fix this as well as part of it, or if not - I will make a new issue about it then so we can handle it in a separate PR.